### DISCUSSION TITLE
Is it possible to have the F flag set to 1 so transaction continues on partial reverts ?

### DISCUSSION THREAD

**Discussion by bscNinja** [Elnaril/uniswap-universal-router-decoder#40] [Q&A]
I tried changing i.e. 0x08 => 0x88 which executes the comands in the chain fine, but the desired effect of partial revert does not work.

Comands.sol

`bytes1 internal constant FLAG_ALLOW_REVERT = 0x80;`

Uniswap Technical Reference

`F`

A single bit flag, that signals whether or not the command should be allowed to revert without the whole transaction failing.

If f is 0 aka false and the command reverts, then the entire transaction will revert and none of the commands will be executed.
If f is 1 aka true and the command reverts, then the transaction will continue, allowing us to achieve partial fills. If using this flag, be careful to include further commands that will remove any funds that could be left unused in the UniversalRouter contract.

**Comment by Elnaril** [comment]
Hello @bscNinja !

This lib does not support partial reverts (yet?), but maybe I could implement it if there is enough interest in it.

> I tried changing i.e. 0x08 => 0x88 which executes the comands in the chain fine, but the desired effect of partial revert does not work.

I have not really played with this flag, but this looks good to me. Hard to tell more without details on the whole transaction.

**Reply by bscNinja** [reply]
Can you elaborate on you playing with this Flag or haven`t you at all ?

I re-deployed the universal router contract locally, removing the revert completely and still do get the same results. I`m starting to think this function is not at all implemented the way uniswap is telling it to be.

**Reply by Elnaril** [reply]
> Can you elaborate on you playing with this Flag or haven`t you at all ?

At some point, I wanted (and even started) to implement it in my lib, but realised it was not very used since I couldn't find a transaction using it, so I turned to more used features/functions. I haven't done/tested any transaction with this flag on, hence the "not really".

> I re-deployed the universal router contract locally, removing the revert completely and still do get the same results. I`m starting to think this function is not at all implemented the way uniswap is telling it to be.

You're tickling my curiosity! :)

**Reply by bscNinja** [reply]
I do understand that it is a not much used function, in my case, imagine multiswapping 100 tokens to get a tax/honeypot result in a single transaction

**Comment by bscNinja** [comment]
Hey Elnaril,

thanks for getting back to me.

Starting to think that either the function in the contract does not work or I`m missing something here. As I understand the Technical Reference 0x80 is a mask, also opened an Isuee on Uniswap Github to get clarification.

Let me know if I can support you in any Way since this functionality is crucial for the success of a Project that I`m working on

**Comment by Elnaril** [comment]
Hey @bscNinja !
So, I've implemented the revert flag and tested it (see branch `v1.1.0.dev0`). I reproduce your issue (see `integration_tests/revert_flag_bsc.py`).
My working hypothesis is that this flag is only for the functions that returns the `success` bool from the UR `dispatch()` method. So for example, `SUDOSWAP`, which is the function cited in the doc.
So, I guess I've implemented the encode part of this flag for nothing ...  😅 At least I've prepared the way for these NFT functions ...
